[ews-build.webkit.org] Convert AnalyzeLayoutTestsResults to new-style

JonWBedard opened the pull request:
#### f80c4133ba61ea386ed66e9964d12fa3de84ff26
<pre>
[ews-build.webkit.org] Convert AnalyzeLayoutTestsResults to new-style
<a href="https://bugs.webkit.org/show_bug.cgi?id=251736">https://bugs.webkit.org/show_bug.cgi?id=251736</a>
rdar://105041006

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(AnalyzeLayoutTestsResults.report_failure): Return deferred value since
_addToLog is asynchronous.
(AnalyzeLayoutTestsResults.report_pre_existing_failures): Return exit code.
(AnalyzeLayoutTestsResults.retry_build): Ditto.
(AnalyzeLayoutTestsResults.send_email_for_new_test_failures): Ditto.
(AnalyzeLayoutTestsResults.run): Renamed from start.
(AnalyzeLayoutTestsResults.start): Renamed to run.
(AnalyzeLayoutTestsResultsRedTree.report_success): Return exit code.
(AnalyzeLayoutTestsResultsRedTree.report_warning): Ditto.
(AnalyzeLayoutTestsResultsRedTree.run):Renamed from start.
(AnalyzeLayoutTestsResultsRedTree.start): Renamed to run.

Canonical link: <a href="https://commits.webkit.org/262104@main">https://commits.webkit.org/262104@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dbbdde5801c7cf105c76a884305c9bc7c869c71d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106185 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15237 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/39020 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115369 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115056 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/110091 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16677 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6459 "Built successfully") | [⏳ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/WPE-WK2-Tests-EWS "Waiting to run tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111945 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12692 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95664 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98390 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94566 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/27307 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/40226 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8483 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28659 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/81909 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/8984 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/5242 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/104994 "Passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14600 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48205 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10520 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/68 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

aj062 reviewed (approved):
r+ assuming this is tested.

JonWBedard committed:
[ews-build.webkit.org] Convert AnalyzeLayoutTestsResults to new-style
https://bugs.webkit.org/show_bug.cgi?id=251736
rdar://105041006

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(AnalyzeLayoutTestsResults.report_failure): Return deferred value since
_addToLog is asynchronous.
(AnalyzeLayoutTestsResults.report_pre_existing_failures): Return exit code.
(AnalyzeLayoutTestsResults.retry_build): Ditto.
(AnalyzeLayoutTestsResults.send_email_for_new_test_failures): Ditto.
(AnalyzeLayoutTestsResults.run): Renamed from start.
(AnalyzeLayoutTestsResults.start): Renamed to run.
(AnalyzeLayoutTestsResultsRedTree.report_success): Return exit code.
(AnalyzeLayoutTestsResultsRedTree.report_warning): Ditto.
(AnalyzeLayoutTestsResultsRedTree.run):Renamed from start.
(AnalyzeLayoutTestsResultsRedTree.start): Renamed to run.

Canonical link: https://commits.webkit.org/262104@main
